subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/resolver/DefaultGradleApiSourcesResolver.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugins.ide.internal.resolver;
+
+import org.gradle.internal.installation.CurrentGradleInstallation;
+import org.gradle.internal.installation.GradleInstallation;
+
+import java.io.File;
+
+public class DefaultGradleApiSourcesResolver implements GradleApiSourcesResolver {
+
+    @Override
+    public File resolveGradleApiSources(File artifact) {
+        GradleInstallation gradleInstallation = CurrentGradleInstallation.get();
+        // TODO: if source dir does not exist, we might want to download a distribution and repackage sources
```

Comment:
Nit: Given this another feature that we aren't sure if we are going to address anytime soon, it would be better to track via an issue instead of a `TODO`. You can mention in the issue this location in the code. It avoid bloating the code with TODOs everywhere.

Reply:
I was thinking I'll address it this week as part of the same issue - I'd say I'm halfway there unless there is something more pressing to shift my attention to.
Agree about TODO - removing it.